### PR TITLE
gtk-doc-stub: add gtk-doc-stub/cci.20181216 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **gtk-doc-stub/cci.201812126**

Needed for running autoreconf on cairo

Depends on :

- [x] https://github.com/conan-io/hooks/pull/343

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/gtk-doc-stub/all/test_package/configure.ac [hunk 1]
```diff
@@ -0,0 +1,9 @@
+AC_INIT([test_package], [1.0])
+
+AC_CONFIG_SRCDIR([configure.ac])
+
+AC_PROG_CC
+
+GTK_DOC_CHECK
```
Comment (madebr) [inline recipes/gtk-doc-stub/all/test_package/configure.ac:7]:
```suggestion
AC_CONFIG_SRCDIR([configure.ac])

GTK_DOC_CHECK
```

File: recipes/gtk-doc-stub/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
+from conans import AutoToolsBuildEnvironment, ConanFile, tools
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class GtkDocStubConan(ConanFile):
+    name = "gtk-doc-stub"
+    homepage = "https://gitlab.gnome.org/GNOME/gtk-doc-stub"
+    description = "Helper scripts for generating GTK documentaiton"
```
Comment (madebr) [inline recipes/gtk-doc-stub/all/conanfile.py:10]:
```suggestion
    description = "Helper scripts for generating GTK documentation"
```

**Comment by jgsogo** [comment]
I think here you will be bitten by the header-onlys _side effect_: now the `test_package` will run on all the platforms (https://github.com/conan-io/conan-center-index/pull/7179)

File: recipes/gtk-doc-stub/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+required_conan_version = ">=1.33.0"
+
+
+class GtkDocStubConan(ConanFile):
+    name = "gtk-doc-stub"
+    homepage = "https://gitlab.gnome.org/GNOME/gtk-doc-stub"
+    description = "Helper scripts for generating GTK documentation"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "GPLv2-or-later"
```
Comment (uilianries) [inline recipes/gtk-doc-stub/all/conanfile.py:13]:
```suggestion
    license = "GPL-2.0-or-later"
```

GPLv2 is valid, but let's try to use the same short name too keep a pattern.

https://spdx.org/licenses/GPL-2.0-or-later.html

**Review by uilianries — APPROVED** [review]

File: recipes/gtk-doc-stub/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+
+class GtkDocStubConan(ConanFile):
+    name = "gtk-doc-stub"
+    homepage = "https://gitlab.gnome.org/GNOME/gtk-doc-stub"
+    description = "Helper scripts for generating GTK documentation"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "GPL-2.0-or-later"
+    topics = ("gtk", "documentation", "gtkdocize")
+    settings = "os"
```
Comment (uilianries) [inline recipes/gtk-doc-stub/all/conanfile.py:15]: Only OS ?

Comment (madebr) [reply recipes/gtk-doc-stub/all/conanfile.py:15]: it's script-only + platform independent. Hence the need for an exception in conan-hooks.
But I see that `self.settings.os` is not used at all.
So I"ll remove that.

File: recipes/gtk-doc-stub/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+    exports_sources = "patches/*"
```
Comment (madebr) [inline recipes/gtk-doc-stub/all/conanfile.py:17]:
```suggestion
    topics = ("gtk", "documentation", "gtkdocize")

    exports_sources = "patches/*"
```

File: recipes/gtk-doc-stub/all/conanfile.py (hunk 4 above)
Comment (madebr) [reply recipes/gtk-doc-stub/all/conanfile.py]: I was wrong :D

File: recipes/gtk-doc-stub/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+class GtkDocStubConan(ConanFile):
+    name = "gtk-doc-stub"
+    homepage = "https://gitlab.gnome.org/GNOME/gtk-doc-stub"
+    description = "Helper scripts for generating GTK documentation"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "GPL-2.0-or-later"
+    topics = ("gtk", "documentation", "gtkdocize")
+
+    exports_sources = "patches/*"
```
Comment (madebr) [inline recipes/gtk-doc-stub/all/conanfile.py:16]:
```suggestion
    topics = ("gtk", "documentation", "gtkdocize")
    settings = "os"

    exports_sources = "patches/*
```

File: recipes/gtk-doc-stub/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+class GtkDocStubConan(ConanFile):
+    name = "gtk-doc-stub"
+    homepage = "https://gitlab.gnome.org/GNOME/gtk-doc-stub"
+    description = "Helper scripts for generating GTK documentation"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "GPL-2.0-or-later"
+    topics = ("gtk", "documentation", "gtkdocize")
+    settings = "os"
+
+    exports_sources = "patches/*
```
Comment (jgsogo) [inline recipes/gtk-doc-stub/all/conanfile.py:17]:
```suggestion
    exports_sources = "patches/*"
```

**Review by jgsogo — APPROVED** [review]
We should run some NLP on commit messages 😄

**Review by prince-chrismc — APPROVED** [review]

**Comment by madebr** [comment]
Hey @uilianries, please give a final review to this one.

**Review by SSE4 — APPROVED** [review]

**Pull request merged**